tds.go

```diff
@@ -756,16 +769,31 @@ func connect(params map[string]string) (res *tdsSession, err error) {
 
 		ips = []net.IP{ip}
 	}
-
-	d := createDialer(p)
-	var conn net.Conn
+	connChan := make(chan net.Conn, len(ips))
+	errChan := make(chan error, len(ips))
 	for _, ip := range ips {
-		addr := fmt.Sprintf("%s:%d", ip, p.port)
-		conn, err = d.Dial("tcp", addr)
-		if err == nil {
-			break
-		}
+		go func(ip net.IP) {
+			d := createDialer(p)
+			addr := fmt.Sprintf("%s:%d", ip, p.port)
+			conn, err := d.Dial("tcp", addr)
+			if err == nil {
+				connChan <- conn
+			} else {
+				errChan <- err
+			}
+		}(ip)
 	}
+	var conn net.Conn
+	select {
+	case conn = <-connChan:
+		// Got a connection to use, close any others
+		go drainDialAttempts(len(ips)-1, connChan, errChan)
+	case <-time.After(p.dial_timeout):
```

Comment:
case err <- errChan is missing here (if everything fails both err and conn are nil).
Since each dial has its own timeout there is no need to handle it here.